From 7a5ab0d4c6b00d834373de33cd87cbafe4eee9c7 Mon Sep 17 00:00:00 2001
From: warped-rudi <r.ihle@s-t.de>
Date: Wed, 5 Aug 2015 20:21:17 +0200
Subject: [PATCH 3/3] video: mxc_edid: Fix handling of multiple audio data
 blocks

The current implemenation was resetting any previouse detected information
at the beginning of a new audio data block. This should not be done, because
CEA861 allows the use of multiple audio data blocks and the last one is not
may not contain the LPCM information we are interrested in.
---
 drivers/video/mxc/mxc_edid.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/video/mxc/mxc_edid.c b/drivers/video/mxc/mxc_edid.c
index 6e2b04c..e27a977 100644
--- a/drivers/video/mxc/mxc_edid.c
+++ b/drivers/video/mxc/mxc_edid.c
@@ -299,6 +299,10 @@ int mxc_edid_parse_ext_blk(unsigned char *edid,
 	}
 
 	if (revision >= 3) {
+		cfg->max_channels = 0;
+		cfg->sample_rates = 0;
+		cfg->sample_sizes = 0;
+
 		/* short desc */
 		DPRINTK("CEA Short desc timmings\n");
 		index++;
@@ -503,10 +507,6 @@ int mxc_edid_parse_ext_blk(unsigned char *edid,
 					u8 audio_format, max_ch, byte1, byte2, byte3;
 
 					i = 0;
-					cfg->max_channels = 0;
-					cfg->sample_rates = 0;
-					cfg->sample_sizes = 0;
-
 					while (i < blklen) {
 						byte1 = edid[index + 1];
 						byte2 = edid[index + 2];
-- 
1.7.9.5

